addons/l10n_in/models/company.py

```diff
@@ -1,6 +1,13 @@
-from odoo import api, fields, models
+from odoo import fields, models, _
+from odoo.exceptions import UserError
 
 class ResCompany(models.Model):
     _inherit = 'res.company'
 
     l10n_in_upi_id = fields.Char(string="UPI Id")
+    l10n_in_active_audit_trail = fields.Boolean(string="Audit Trail", default=False, help="Once you activate audit trail it can't be deactivated.")
+
+    def write(self, vals):
+        if 'l10n_in_active_audit_trail' in vals and self._origin.l10n_in_active_audit_trail and not vals['l10n_in_active_audit_trail']:
```

Comment:
do we need an ._origin in writes?  Also, self can be multiple records...  It is weird as well that you could be change put the setting and you won't be able to set it back after.  We could just have a tracking True.  @jva-odoo

Reply:
it's mandatory that once you activate it no one can change it.
![image](https://github.com/odoo/odoo/assets/34996524/36343518-b8f7-4da3-b7ee-a8576ee48848)

Reply:
![Audit Trail Notification](https://github.com/odoo/odoo/assets/93132516/4f940028-baf4-42c6-9748-66170fd47ae1)

Reply:
@jco-odoo

Reply:
https://drive.google.com/file/d/1ahIAJZDAG-dh8YlD74DkERDQ06QXQeMm/view?pli=1

Full Notification

Reply:
As this Notification is issued to Ministry of Corporate Affairs ( Governing body of Private, Public & Limited Liability companies, so Applicability of this Audit Trail would only limited to such nature of companies.

where in if organisation is Proprietary, Partnership, Hindu Undivided Family, Trust, NGO, Association of Person than this Audit Trail notification will not apply.

Reply:
@tsb-odoo It makes more sense with this explanation ^